### DISCUSSION TITLE
NXP MKE13Z7 On Zephyr RTOS?

### DISCUSSION THREAD

**Discussion by andrea-cpu96** [zephyrproject-rtos/zephyr#79758] [Q&A] [answered]
I have to create a project for our board based on a mke13z7 micocontroller. I saw that the 3.7 version of Zephyr supports the mke1xz family, but it seems that actually only the mke15z7 and mke17z7 are supported and not the mke13z7.

I'm new in Zephyr and if possible I would prefer not to create a new custom SoC for the mke13z7, do you know if there is a sort of "workaround" for this problem?

Maybe I can use the configurations for the mke17 instead, because this microcontroller seems to be very similar to the mke13, but I'm not sure this is the right way.

**Comment by DerekSnell** [comment, marked as answer]
Hi @andrea-cpu96 ,

Yes, as you found, the KE17Z7 SOC is supported in Zephyr, but not the KE13Z7.  They are very similar devices, and part of the same family.  Zephyr has direct support for boards, and SOCs are indirectly supported by those boards.  NXP typically puts the superset device of a family on the development board, which in this case is the [FRDM-KE17Z](https://docs.zephyrproject.org/latest/boards/nxp/frdm_ke17z/doc/index.html).  The KE13Z7 SOC is very similar to the superset device on that board, but does not have all the features/memory/pins/etc..

The formal process in Zephyr is to clone the existing SOC, and then modify that new SOC to change the features that differ from the superset.  And then use that new SOC in your custom board.  But because these SOCs are in the same family and very similar, I think you can re-use the existing KE17Z7 SOC on your custom board.  

Since the existing KE17Z7 will enable more features/memory than your SOC, your board files or app will need to limit those features.  For example, your board DTS can change the memory sizes if different, and can disable peripherals not available on your SOC.  The pin mux options likely have some differences, so you should also review the pinctrl settings on your custom board, and ensure they match the KE13Z7 pin out.

We also noticed you posted this same question on [Reddit](https://www.reddit.com/r/embedded/comments/1g1yjjm/nxp_mke13z7_on_zephyr_rtos/), and our friends at Golioth pointed you to their very detailed and helpful [blog on creating a custom board](https://blog.golioth.io/how-to-add-a-custom-board-definition-to-a-zephyr-application/). 

Let us know how this goes with your custom board.  Best regards
